File: NDPCommon/src/main/java/com/slender/utils/CacheToolkit.java
package com.slender.utils;

public final class CacheToolkit {
    private CacheToolkit(){}

    public static String getKey(Class<?> clazz, Long id){
        String[] split = clazz.getName().split("\\.");
        String name = split[split.length - 1];
        return name + ":" + id;
    }

    public static String getRemoveKey(Class<?> clazz, Long id){
        String[] split = clazz.getName().split("\\.");
        String name = split[split.length - 1];
        return name.substring(0, name.length() - 11) + ":" + id;
    }
}

File: NDPCommon/src/main/java/com/slender/utils/CacheToolkit.kt
package com.slender.utils

object CacheToolkit {
    fun getKey(clazz: Class<*>, id: Long) = "${clazz.simpleName}:$id"

    fun getRemoveKey(clazz: Class<*>, id: Long) = "${clazz.simpleName.removeSuffix("ServiceImpl")}:$id"
}
